aws-s3/spring-cloud-starter-stream-source-s3/src/main/java/org/springframework/cloud/stream/app/s3/source/AmazonS3SourceConfiguration.java

```diff
@@ -57,7 +57,7 @@
 	private ResourceIdResolver resourceIdResolver;
 
 	@Bean
-	public TriggerProperties triggerProperties() {
+	public static TriggerProperties triggerProperties() {
```

Comment:
Well, that maybe a fact why Spring Boot uses `@EnableConfigurationProperties`. See their `EnableConfigurationPropertiesImportSelector`, where they do exactly this:
````java
MultiValueMap<String, Object> attributes = metadata
			.getAllAnnotationAttributes(
					EnableConfigurationProperties.class.getName(), false);
List<Class<?>> types = collectClasses(attributes.get("value"));
for (Class<?> type : types) {
	String prefix = extractPrefix(type);
	String name = (StringUtils.hasText(prefix) ? prefix + "-" + type.getName()
				: type.getName());
	if (!registry.containsBeanDefinition(name)) {
			registerBeanDefinition(registry, type, name);
	}
}
...
BeanDefinitionBuilder builder = BeanDefinitionBuilder
					.genericBeanDefinition(type);
AbstractBeanDefinition beanDefinition = builder.getBeanDefinition();
registry.registerBeanDefinition(name, beanDefinition);
````

`static` really makes sense here and doesn't hurt. Although it is [intended](http://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-java-injecting-imported-beans) (see important not starting with "Make sure that ...") just like for `BPP`, but any "early BeanFactory access" (and I guess SCDF does it) may cause that `BeanCurrentlyInCreationException` syndrome.

But yeah... Extra simple `@Configuration` class will help to overcome the issue. Anyway doesn't look kosher since `@ConfigurationProperties` are intended to be pretty simple POJO and be based only on the properties from the `Environment`.